### PR TITLE
feat(ingestion) dbt: Fixing issue with strip_user_ids_from_email and adding owner_naming_pattern

### PR DESCRIPTION
**Pull request by arunvasudevan**
## Checklist
- [X] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [X] Tests for the changes have been added/updated (if applicable)
- [X] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.

Config `strip_user_ids_from_email` has not been used currently. Added fix to apply the config.

Also, added `owner_naming_pattern` regex to parse the owner name using the regex supplied

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/dbt.py [hunk 1]
```diff
@@ -1026,9 +1016,18 @@ def _aggregate_owners(
     ) -> List[OwnerClass]:
         owner_list: List[OwnerClass] = []
         if node.owner:
+            owner = node.owner
+            if self.config.owner_naming_pattern:
+                found = re.search(re.compile(self.config.owner_naming_pattern), owner)
+                if found:
+                    owner = found.group(2)
```
Comment (rslanka) [inline metadata-ingestion/src/datahub/ingestion/source/dbt.py:1023]: Hi @arunvasudevan, this is making an assumption that (1) there are at least 2 parenthesized groups in the user provided owner naming pattern and (2) the second one happens to be the owner. If these 2 conditions are not met, we will either get an IndexError or an incorrect value for the owner.

My recommendation is to use (?P\<name\>...) syntax, where name=owner, and use found.group('owner') to extract the owner. The documentation for the pattern also needs to reflect the same. Please refer to [Match Objects documentation](https://docs.python.org/3/library/re.html#match-objects) for details.

**Review by rslanka — CHANGES_REQUESTED** [review]
Change requested.

File: metadata-ingestion/src/datahub/ingestion/source/dbt.py (hunk 1 above)
Comment (arunvasudevan) [reply metadata-ingestion/src/datahub/ingestion/source/dbt.py]: @rslanka Updated code changes based on the suggestion. Take a look and let me know your comments

Comment (arunvasudevan) [reply metadata-ingestion/src/datahub/ingestion/source/dbt.py]: @rslanka All checks passed as well. Let me know your thoughts

**Review by rslanka — APPROVED** [review]
LGTM!

**Review by rslanka — APPROVED** [review]
LGTM!

**Pull request merged**